stl/inc/ranges

```diff
@@ -4748,6 +4748,566 @@ namespace ranges {
+            }
+
+            constexpr _Iterator& operator++() noexcept(
+                noexcept(_RANGES advance(_Current, _Count, _End))) /* strengthened */ {
+#if _ITERATOR_DEBUG_LEVEL != 0
+                _STL_VERIFY(_Current != _End, "cannot increment chunk_view end iterator");
+#endif // _ITERATOR_DEBUG_LEVEL != 0
+                _Missing = _RANGES advance(_Current, _Count, _End);
+                return *this;
+            }
+
+            constexpr _Iterator operator++(int) noexcept(is_nothrow_copy_constructible_v<_Base_iter>&& noexcept(
+                _RANGES advance(_Current, _Count, _End))) /* strengthened */ {
+                auto _Tmp = *this;
+                ++*this;
+                return _Tmp;
+            }
+
+            constexpr _Iterator& operator--() noexcept(
+                noexcept(_RANGES advance(_Current, _Missing - _Count))) /* strengthened */
+                requires bidirectional_range<_Base> {
+                _RANGES advance(_Current, _Missing - _Count);
+                _Missing = 0;
+                return *this;
+            }
+
+            constexpr _Iterator operator--(int) noexcept(is_nothrow_copy_constructible_v<_Base_iter>&& noexcept(
+                _RANGES advance(_Current, _Missing - _Count))) /* strengthened */
+                requires bidirectional_range<_Base> {
+                auto _Tmp = *this;
+                --*this;
+                return _Tmp;
+            }
+
+            constexpr _Iterator& operator+=(const difference_type _Off) noexcept(
+                noexcept(_RANGES advance(_Current, _Count* _Off, _End))) /* strengthened */
+                requires random_access_range<_Base> {
+                if (_Off > 0) {
+#if _ITERATOR_DEBUG_LEVEL != 0
+                    _STL_VERIFY(_RANGES distance(_Current, _End) > _Count * (_Off - 1),
+                        "cannot advance chunk_view iterator past end");
+#endif // _ITERATOR_DEBUG_LEVEL != 0
+                    _Missing = _RANGES advance(_Current, _Count * _Off, _End);
+                } else if (_Off < 0) {
+                    _RANGES advance(_Current, _Count * _Off + _Missing);
+                    _Missing = 0;
+                }
+                return *this;
+            }
+
+            constexpr _Iterator& operator-=(const difference_type _Off) noexcept(
+                noexcept(_RANGES advance(_Current, -_Count* _Off, _End))) /* strengthened */
+                requires random_access_range<_Base> {
+                if (_Off < 0) {
+#if _ITERATOR_DEBUG_LEVEL != 0
+                    _STL_VERIFY(_RANGES distance(_Current, _End) > -_Count * (_Off - 1),
+                        "cannot advance chunk_view iterator past end");
+#endif // _ITERATOR_DEBUG_LEVEL != 0
+                    _Missing = _RANGES advance(_Current, -_Count * _Off, _End);
+                } else if (_Off > 0) {
+                    _RANGES advance(_Current, -_Count * _Off + _Missing);
+                    _Missing = 0;
+                }

```

Comment:
any reason to not just do `operator+=(-_Off)`? no change requested

Reply:
Because debug performance makes @BillyONeal happy

Reply:
![image](https://user-images.githubusercontent.com/1544943/167195540-17c5f179-cef8-433e-9817-08b2d391ac0e.png)